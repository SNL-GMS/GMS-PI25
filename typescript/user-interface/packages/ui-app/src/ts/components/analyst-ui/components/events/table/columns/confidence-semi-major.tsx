import type { ColumnDefinition } from '@gms/ui-core-components';
import { EventsColumn } from '@gms/ui-state';
import type { ICellRendererParams, IHeaderParams } from 'ag-grid-community';

import { decimalPrecisionValueFormatter } from '~common-ui/common/table-utils';

import type { EventRow } from '../../types';
import { eventColumnDisplayStrings } from '../../types';

/**
 * Defines the confidence semi major column definition
 *
 * @param columnsToDisplayMap a map that specifies if the column is visible
 * @returns the column definition
 */
export const confidenceSemiMajorColumnDef: ColumnDefinition<
  EventRow,
  unknown,
  number,
  ICellRendererParams,
  IHeaderParams
> = {
  headerName: eventColumnDisplayStrings.get(EventsColumn.confidenceSemiMajorAxis),
  field: EventsColumn.confidenceSemiMajorAxis,
  headerTooltip: 'Confidence semi-major axis in kilometers',
  width: 130,
  valueFormatter: params =>
    decimalPrecisionValueFormatter<EventRow>(params, EventsColumn.confidenceSemiMajorAxis, 2)
};
