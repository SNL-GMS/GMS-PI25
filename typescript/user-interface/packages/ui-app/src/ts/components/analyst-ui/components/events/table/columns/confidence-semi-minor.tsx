import type { ColumnDefinition } from '@gms/ui-core-components';
import { EventsColumn } from '@gms/ui-state';
import type { ICellRendererParams, IHeaderParams } from 'ag-grid-community';

import { decimalPrecisionValueFormatter } from '~common-ui/common/table-utils';

import type { EventRow } from '../../types';
import { eventColumnDisplayStrings } from '../../types';

/**
 * Defines the confidence semi minor column definition
 *
 * @param columnsToDisplayMap a map that specifies if the column is visible
 * @returns the column definition
 */
export const confidenceSemiMinorColumnDef: ColumnDefinition<
  EventRow,
  unknown,
  number,
  ICellRendererParams,
  IHeaderParams
> = {
  headerName: eventColumnDisplayStrings.get(EventsColumn.confidenceSemiMinorAxis),
  field: EventsColumn.confidenceSemiMinorAxis,
  headerTooltip: 'Confidence semi-minor axis in kilometers',
  width: 130,
  valueFormatter: params =>
    decimalPrecisionValueFormatter<EventRow>(params, EventsColumn.confidenceSemiMinorAxis, 2)
};
