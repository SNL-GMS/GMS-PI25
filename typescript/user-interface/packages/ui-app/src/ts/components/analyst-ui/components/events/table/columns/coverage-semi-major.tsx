import type { ColumnDefinition } from '@gms/ui-core-components';
import { EventsColumn } from '@gms/ui-state';
import type { ICellRendererParams, IHeaderParams } from 'ag-grid-community';

import { decimalPrecisionValueFormatter } from '~common-ui/common/table-utils';

import type { EventRow } from '../../types';
import { eventColumnDisplayStrings } from '../../types';

/**
 * Defines the coverage semi major column definition
 *
 * @param columnsToDisplayMap a map that specifies if the column is visible
 * @returns the column definition
 */
export const coverageSemiMajorColumnDef: ColumnDefinition<
  EventRow,
  unknown,
  number,
  ICellRendererParams,
  IHeaderParams
> = {
  headerName: eventColumnDisplayStrings.get(EventsColumn.coverageSemiMajorAxis),
  field: EventsColumn.coverageSemiMajorAxis,
  headerTooltip: 'Coverage semi-major axis in kilometers',
  width: 130,
  valueFormatter: params =>
    decimalPrecisionValueFormatter<EventRow>(params, EventsColumn.coverageSemiMajorAxis, 2)
};
