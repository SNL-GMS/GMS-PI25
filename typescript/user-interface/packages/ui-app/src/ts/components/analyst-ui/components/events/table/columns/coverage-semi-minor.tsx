import type { ColumnDefinition } from '@gms/ui-core-components';
import { EventsColumn } from '@gms/ui-state';
import type { ICellRendererParams, IHeaderParams } from 'ag-grid-community';

import { decimalPrecisionValueFormatter } from '~common-ui/common/table-utils';

import type { EventRow } from '../../types';
import { eventColumnDisplayStrings } from '../../types';

/**
 * Defines the coverage semi minor column definition
 *
 * @param columnsToDisplayMap a map that specifies if the column is visible
 * @returns the column definition
 */
export const coverageSemiMinorColumnDef: ColumnDefinition<
  EventRow,
  unknown,
  number,
  ICellRendererParams,
  IHeaderParams
> = {
  headerName: eventColumnDisplayStrings.get(EventsColumn.coverageSemiMinorAxis),
  field: EventsColumn.coverageSemiMinorAxis,
  headerTooltip: 'Coverage semi-minor axis in kilometers',
  width: 130,
  valueFormatter: params =>
    decimalPrecisionValueFormatter<EventRow>(params, EventsColumn.coverageSemiMinorAxis, 2)
};
