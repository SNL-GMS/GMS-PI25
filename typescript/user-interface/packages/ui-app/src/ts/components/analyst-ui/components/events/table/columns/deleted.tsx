import type { ColumnDefinition } from '@gms/ui-core-components';
import { EventsColumn } from '@gms/ui-state';
import type { ICellRendererParams, IHeaderParams } from 'ag-grid-community';

import { booleanValueFormatter } from '~common-ui/common/table-utils';

import type { EventRow } from '../../types';
import { eventColumnDisplayStrings } from '../../types';

/**
 * Defines the deleted column definition
 *
 * @param columnsToDisplayMap a map that specifies if the column is visible
 * @returns the column definition
 */
export const deletedColumnDef: ColumnDefinition<
  EventRow,
  unknown,
  boolean,
  ICellRendererParams,
  IHeaderParams
> = {
  headerName: eventColumnDisplayStrings.get(EventsColumn.deleted),
  field: EventsColumn.deleted,
  headerTooltip: 'Deleted',
  valueFormatter: params => booleanValueFormatter<EventRow>(params, EventsColumn.deleted)
};
