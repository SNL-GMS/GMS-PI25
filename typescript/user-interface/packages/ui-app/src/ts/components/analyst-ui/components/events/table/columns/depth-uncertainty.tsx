import type { Depth } from '@gms/common-model/lib/event';
import { setDecimalPrecision } from '@gms/common-util';
import type {
  CellRendererParams,
  ColumnDefinition,
  ValueFormatterParams,
  ValueGetterParams
} from '@gms/ui-core-components';
import { EventsColumn } from '@gms/ui-state';
import type { ICellRendererParams, IHeaderParams } from 'ag-grid-community';

import type { EventRow } from '../../types';
import { eventColumnDisplayStrings } from '../../types';

/**
 * @returns Formatted string, for depthUncertainty when it exists
 */
function depthUncertaintyValueFormatter(
  params: ValueFormatterParams<
    EventRow,
    unknown,
    string,
    CellRendererParams<EventRow, unknown, Depth, ICellRendererParams, IHeaderParams>,
    IHeaderParams
  >
): string {
  return params.data.depthKm.uncertainty
    ? setDecimalPrecision(params.data.depthKm.uncertainty, 3)
    : '';
}

/**
 * Custom valueGetter so that this column can be sorted on depthKm only.
 *
 * @returns depthKm as a number
 */
function timeUncertaintyValueGetter(
  params: ValueGetterParams<
    EventRow,
    unknown,
    string,
    CellRendererParams<EventRow, unknown, Depth, ICellRendererParams, IHeaderParams>,
    IHeaderParams
  >
): number {
  return params.data.depthKm.value;
}

/**
 * Defines the depth column definition
 *
 * @param columnsToDisplayMap a map that specifies if the column is visible
 * @returns the column definition
 */
export const depthUncertaintyColumnDef: ColumnDefinition<
  EventRow,
  unknown,
  string,
  ICellRendererParams,
  IHeaderParams
> = {
  headerName: eventColumnDisplayStrings.get(EventsColumn.depthUncertainty),
  field: EventsColumn.depthUncertainty,
  headerTooltip: 'Depth standard deviation',
  width: 150,
  valueFormatter: depthUncertaintyValueFormatter,
  valueGetter: timeUncertaintyValueGetter
};
