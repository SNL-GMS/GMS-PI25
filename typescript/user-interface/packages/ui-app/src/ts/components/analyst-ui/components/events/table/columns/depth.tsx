import type { Depth } from '@gms/common-model/lib/event';
import { setDecimalPrecision } from '@gms/common-util';
import type {
  CellRendererParams,
  ColumnDefinition,
  ValueFormatterParams,
  ValueGetterParams
} from '@gms/ui-core-components';
import { EventsColumn } from '@gms/ui-state';
import type { ICellRendererParams, IHeaderParams } from 'ag-grid-community';

import { medCellWidthPx } from '~common-ui/common/table-types';

import type { EventRow } from '../../types';
import { eventColumnDisplayStrings } from '../../types';

/**
 * @returns Formatted string
 */
function depthValueFormatter(
  params: ValueFormatterParams<
    EventRow,
    unknown,
    Depth,
    CellRendererParams<EventRow, unknown, Depth, ICellRendererParams, IHeaderParams>,
    IHeaderParams
  >
): string {
  return setDecimalPrecision(params.data.depthKm.value, 3);
}

/**
 * Custom valueGetter so that this column can be sorted on depthKm only.
 *
 * @returns depthKm as a number
 */
function timeValueGetter(
  params: ValueGetterParams<
    EventRow,
    unknown,
    string,
    CellRendererParams<EventRow, unknown, Depth, ICellRendererParams, IHeaderParams>,
    IHeaderParams
  >
): number {
  return params.data.depthKm.value;
}

/**
 * Defines the depth column definition
 *
 * @param columnsToDisplayMap a map that specifies if the column is visible
 * @returns the column definition
 */
export const depthColumnDef: ColumnDefinition<
  EventRow,
  unknown,
  Depth,
  ICellRendererParams,
  IHeaderParams
> = {
  headerName: eventColumnDisplayStrings.get(EventsColumn.depthKm),
  field: EventsColumn.depthKm,
  headerTooltip: 'Depth in kilometers',
  width: medCellWidthPx,
  valueFormatter: depthValueFormatter,
  valueGetter: timeValueGetter
};
