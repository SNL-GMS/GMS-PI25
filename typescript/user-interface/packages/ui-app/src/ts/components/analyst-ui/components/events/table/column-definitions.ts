import { EventsColumn } from '@gms/ui-state';

import { getConflictColumnDef } from '~analyst-ui/common/table/conflict-marker/conflict-marker-column-def';
import { getDirtyDotColumnDef } from '~analyst-ui/common/table/dirty-dot/dirty-dot-column-def';

import { eventColumnDisplayStrings } from '../types';
import { activeAnalystsColumnDef } from './columns/active-analysts';
import { confidenceSemiMajorColumnDef } from './columns/confidence-semi-major';
import { confidenceSemiMinorColumnDef } from './columns/confidence-semi-minor';
import { coverageSemiMajorColumnDef } from './columns/coverage-semi-major';
import { coverageSemiMinorColumnDef } from './columns/coverage-semi-minor';
import { deletedColumnDef } from './columns/deleted';
import { depthColumnDef } from './columns/depth';
import { depthUncertaintyColumnDef } from './columns/depth-uncertainty';
import { latitudeColumnDef } from './columns/latitude';
import { longitudeColumnDef } from './columns/longitude';
import { mbColumnDef } from './columns/mb';
import { mlColumnDef } from './columns/ml';
import { msColumnDef } from './columns/ms';
import { numberAssociatedColumnDef } from './columns/number-associated';
import { numberDefiningColumnDef } from './columns/number-defining';
import { observationsStandardDeviationColumnDef } from './columns/observerations-standard-dev';
import { preferredColumnDef } from './columns/preferred';
import { regionColumnDef } from './columns/region';
import { rejectedColumnDef } from './columns/rejected';
import { statusColumnDef } from './columns/status';
import { timeColumnDef } from './columns/time';
import { timeUncertaintyColumnDef } from './columns/time-uncertainty';

/**
 * @returns List of column definition objects for the Events Table
 */
export const getEventsTableColumnDefs = () => {
  return [
    getDirtyDotColumnDef(
      eventColumnDisplayStrings.get(EventsColumn.unsavedChanges),
      EventsColumn.unsavedChanges,
      eventColumnDisplayStrings.get(EventsColumn.unsavedChanges)
    ),
    getConflictColumnDef(
      eventColumnDisplayStrings.get(EventsColumn.conflict),
      EventsColumn.conflict,
      eventColumnDisplayStrings.get(EventsColumn.conflict)
    ),
    timeColumnDef,
    timeUncertaintyColumnDef,
    latitudeColumnDef,
    longitudeColumnDef,
    depthColumnDef,
    depthUncertaintyColumnDef,
    mbColumnDef,
    msColumnDef,
    mlColumnDef,
    numberAssociatedColumnDef,
    numberDefiningColumnDef,
    observationsStandardDeviationColumnDef,
    coverageSemiMajorColumnDef,
    coverageSemiMinorColumnDef,
    confidenceSemiMajorColumnDef,
    confidenceSemiMinorColumnDef,
    regionColumnDef,
    activeAnalystsColumnDef,
    preferredColumnDef,
    statusColumnDef,
    rejectedColumnDef,
    deletedColumnDef
  ];
};
