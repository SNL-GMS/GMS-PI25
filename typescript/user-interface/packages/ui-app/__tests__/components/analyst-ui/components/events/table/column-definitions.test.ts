/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable @typescript-eslint/no-magic-numbers */
import { location } from '@gms/common-model/__tests__/__data__';
import type {
  CellRendererParams,
  TooltipParams,
  ValueFormatterParams
} from '@gms/ui-core-components';

import { deletedColumnDef } from '~analyst-ui/components/events/table/columns/deleted';
import { depthUncertaintyColumnDef } from '~analyst-ui/components/events/table/columns/depth-uncertainty';
import { numberAssociatedColumnDef } from '~analyst-ui/components/events/table/columns/number-associated';
import { numberDefiningColumnDef } from '~analyst-ui/components/events/table/columns/number-defining';
import { timeUncertaintyColumnDef } from '~analyst-ui/components/events/table/columns/time-uncertainty';

import { getEventsTableColumnDefs } from '../../../../../../src/ts/components/analyst-ui/components/events/table/column-definitions';
import type { EventValueGetterParams } from '../../../../../../src/ts/components/analyst-ui/components/events/table/columns/active-analysts';
import {
  activeAnalystsColumnDef,
  activeAnalystsTooltipValueGetter,
  activeAnalystsValueFormatter,
  activeAnalystsValueGetter
} from '../../../../../../src/ts/components/analyst-ui/components/events/table/columns/active-analysts';
import { confidenceSemiMajorColumnDef } from '../../../../../../src/ts/components/analyst-ui/components/events/table/columns/confidence-semi-major';
import { confidenceSemiMinorColumnDef } from '../../../../../../src/ts/components/analyst-ui/components/events/table/columns/confidence-semi-minor';
import { coverageSemiMajorColumnDef } from '../../../../../../src/ts/components/analyst-ui/components/events/table/columns/coverage-semi-major';
import { coverageSemiMinorColumnDef } from '../../../../../../src/ts/components/analyst-ui/components/events/table/columns/coverage-semi-minor';
import { depthColumnDef } from '../../../../../../src/ts/components/analyst-ui/components/events/table/columns/depth';
import { latitudeColumnDef } from '../../../../../../src/ts/components/analyst-ui/components/events/table/columns/latitude';
import { longitudeColumnDef } from '../../../../../../src/ts/components/analyst-ui/components/events/table/columns/longitude';
import { mbColumnDef } from '../../../../../../src/ts/components/analyst-ui/components/events/table/columns/mb';
import { mlColumnDef } from '../../../../../../src/ts/components/analyst-ui/components/events/table/columns/ml';
import { msColumnDef } from '../../../../../../src/ts/components/analyst-ui/components/events/table/columns/ms';
import { preferredColumnDef } from '../../../../../../src/ts/components/analyst-ui/components/events/table/columns/preferred';
import { regionColumnDef } from '../../../../../../src/ts/components/analyst-ui/components/events/table/columns/region';
import { statusColumnDef } from '../../../../../../src/ts/components/analyst-ui/components/events/table/columns/status';
import { timeColumnDef } from '../../../../../../src/ts/components/analyst-ui/components/events/table/columns/time';
import type { EventRow } from '../../../../../../src/ts/components/analyst-ui/components/events/types';
import { dummyData } from '../event-table-mock-data';

describe('Event Table Column Defs', () => {
  it('is exported', () => {
    expect(activeAnalystsValueGetter).toBeDefined();
    expect(activeAnalystsTooltipValueGetter).toBeDefined();
    expect(activeAnalystsValueFormatter).toBeDefined();
    expect(activeAnalystsColumnDef).toBeDefined();
    expect(confidenceSemiMajorColumnDef).toBeDefined();
    expect(confidenceSemiMinorColumnDef).toBeDefined();
    expect(coverageSemiMajorColumnDef).toBeDefined();
    expect(coverageSemiMinorColumnDef).toBeDefined();
    expect(depthColumnDef).toBeDefined();
    expect(depthUncertaintyColumnDef).toBeDefined();
    expect(latitudeColumnDef).toBeDefined();
    expect(longitudeColumnDef).toBeDefined();
    expect(mbColumnDef).toBeDefined();
    expect(mlColumnDef).toBeDefined();
    expect(msColumnDef).toBeDefined();
    expect(preferredColumnDef).toBeDefined();
    expect(regionColumnDef).toBeDefined();
    expect(timeColumnDef).toBeDefined();
    expect(timeUncertaintyColumnDef).toBeDefined();
    expect(statusColumnDef).toBeDefined();
    expect(numberAssociatedColumnDef).toBeDefined();
    expect(numberDefiningColumnDef).toBeDefined();
    expect(deletedColumnDef).toBeDefined();
    expect(getEventsTableColumnDefs).toBeDefined();
  });

  it('has column definitions', () => {
    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
    expect(getEventsTableColumnDefs()).toHaveLength(24);
    expect(getEventsTableColumnDefs()).toMatchSnapshot();
  });

  it('has active analysts value getters', () => {
    const params: EventValueGetterParams = {
      getValue(field: string) {
        dummyData.activeAnalysts.join(', ');
      },
      node: undefined,
      data: dummyData,
      colDef: undefined,
      column: undefined,
      api: undefined,
      columnApi: undefined,
      context: undefined
    };

    const formattedParams: ValueFormatterParams<
      EventRow,
      unknown,
      string,
      CellRendererParams<any, any, any, any, any>,
      unknown
    > = {
      value: '',
      node: undefined,
      data: undefined,
      colDef: undefined,
      column: undefined,
      api: undefined,
      columnApi: undefined,
      context: undefined
    };

    const tooltipParams: TooltipParams = {
      ...params,
      location: 'cell'
    };

    params.data = { ...params.data, activeAnalysts: undefined };

    expect(activeAnalystsValueGetter(params)).toMatchInlineSnapshot(`undefined`);
    expect(activeAnalystsValueFormatter(formattedParams)).toMatchInlineSnapshot(`""`);

    params.data = { ...params.data, activeAnalysts: [] };

    expect(activeAnalystsValueGetter(params)).toMatchInlineSnapshot(`""`);
    expect(activeAnalystsValueFormatter(formattedParams)).toMatchInlineSnapshot(`""`);

    params.data = { ...params.data, activeAnalysts: ['Chillas'] };

    expect(activeAnalystsValueGetter(params)).toMatchInlineSnapshot(`"Chillas"`);
    expect(activeAnalystsValueFormatter(formattedParams)).toMatchInlineSnapshot(`""`);

    params.data = { ...params.data, activeAnalysts: dummyData.activeAnalysts };

    expect(activeAnalystsValueGetter(params)).toMatchInlineSnapshot(
      `"Chillas, Echidnas, I&T, Platform, SMEs"`
    );
    expect(activeAnalystsValueFormatter(formattedParams)).toMatchInlineSnapshot(`""`);
    expect(activeAnalystsTooltipValueGetter(tooltipParams)).toMatchInlineSnapshot(`undefined`);
  });
  it('has data formatters', () => {
    const formatterParams: any = {
      data: dummyData,
      api: undefined,
      colDef: undefined,
      column: undefined,
      addRenderedRowListener: undefined,
      columnApi: undefined,
      context: undefined,
      eGridCell: undefined,
      eParentOfValue: undefined,
      node: undefined,
      value: dummyData.activeAnalysts.join(', '),
      refreshCell: undefined,
      rowIndex: undefined,
      formatValue: undefined,
      getValue: undefined,
      setValue: undefined,
      valueFormatted: undefined,
      $scope: undefined
    };

    const mockSetDecimalPrecision = jest.fn(() => '0.0');

    jest.mock('@gms/ui-core-components', () => ({
      setDecimalPrecision: mockSetDecimalPrecision
    }));

    let testValue = '';

    // lat
    const latColDefFormatter = getEventsTableColumnDefs()[4].valueFormatter;
    if (typeof latColDefFormatter === 'function') {
      testValue = latColDefFormatter(formatterParams);
    }
    expect(testValue).toMatch(`${location.latitudeDegrees}`);

    // lon
    const lonDefFormatter = getEventsTableColumnDefs()[5].valueFormatter;
    if (typeof lonDefFormatter === 'function') {
      testValue = lonDefFormatter(formatterParams);
    }
    expect(testValue).toMatch(`${location.longitudeDegrees}`);
    // depth
    const depthDefFormatter = getEventsTableColumnDefs()[6].valueFormatter;
    if (typeof depthDefFormatter === 'function') {
      testValue = depthDefFormatter(formatterParams);
    }
    expect(testValue).toMatch(`${location.depthKm}`);

    // mb
    const mbDefFormatter = getEventsTableColumnDefs()[8].valueFormatter;
    if (typeof mbDefFormatter === 'function') {
      testValue = mbDefFormatter(formatterParams);
    }
    expect(testValue).toMatch('5.2');

    // ms
    const msDefFormatter = getEventsTableColumnDefs()[9].valueFormatter;
    if (typeof msDefFormatter === 'function') {
      testValue = msDefFormatter(formatterParams);
    }
    expect(testValue).toMatch('4.9');

    // ml
    const mlDefFormatter = getEventsTableColumnDefs()[10].valueFormatter;
    if (typeof mlDefFormatter === 'function') {
      testValue = mlDefFormatter(formatterParams);
    }
    expect(testValue).toMatch('5.0');

    // number associated
    const numberAssociatedDefFormatter = getEventsTableColumnDefs()[11].valueFormatter;
    if (typeof numberAssociatedDefFormatter === 'function') {
      testValue = numberAssociatedDefFormatter(formatterParams);
    }
    expect(testValue).toMatch('0');

    // number defining
    const numberDefiningDefFormatter = getEventsTableColumnDefs()[12].valueFormatter;
    if (typeof numberDefiningDefFormatter === 'function') {
      testValue = numberDefiningDefFormatter(formatterParams);
    }
    expect(testValue).toMatch('0');

    // coverage major axis
    const coverageSemiMajorDefFormatter = getEventsTableColumnDefs()[14].valueFormatter;
    if (typeof coverageSemiMajorDefFormatter === 'function') {
      testValue = coverageSemiMajorDefFormatter(formatterParams);
    }
    expect(testValue).toMatch('820.24');

    // coverage minor axis
    const coverageSemiMinorDefFormatter = getEventsTableColumnDefs()[15].valueFormatter;
    if (typeof coverageSemiMinorDefFormatter === 'function') {
      testValue = coverageSemiMinorDefFormatter(formatterParams);
    }
    expect(testValue).toMatch('677.49');

    // confidence major axis
    const confidenceSemiMajorDefFormatter = getEventsTableColumnDefs()[16].valueFormatter;
    if (typeof confidenceSemiMajorDefFormatter === 'function') {
      testValue = confidenceSemiMajorDefFormatter(formatterParams);
    }
    expect(testValue).toMatch('120.25');

    // confidence minor axis
    const confidenceSemiMinorDefFormatter = getEventsTableColumnDefs()[17].valueFormatter;
    if (typeof confidenceSemiMinorDefFormatter === 'function') {
      testValue = confidenceSemiMinorDefFormatter(formatterParams);
    }
    expect(testValue).toMatch('67.41');
  });

  it('has filter value getters', () => {
    const formatterParams: any = {
      data: dummyData,
      api: undefined,
      colDef: undefined,
      column: undefined,
      addRenderedRowListener: undefined,
      columnApi: undefined,
      context: undefined,
      eGridCell: undefined,
      eParentOfValue: undefined,
      node: undefined,
      value: dummyData.activeAnalysts.join(', '),
      refreshCell: undefined,
      rowIndex: undefined,
      formatValue: undefined,
      getValue: undefined,
      setValue: undefined,
      valueFormatted: undefined,
      $scope: undefined
    };

    const mockSetDecimalPrecision = jest.fn(() => '0.0');

    jest.mock('@gms/ui-core-components', () => ({
      setDecimalPrecision: mockSetDecimalPrecision
    }));

    let testValue = '';

    // lat
    const latColDefFormatter = getEventsTableColumnDefs()[4].filterValueGetter;
    if (typeof latColDefFormatter === 'function') {
      testValue = latColDefFormatter(formatterParams);
    }
    expect(testValue).toMatch(`${location.latitudeDegrees}`);

    // lon
    const lonDefFormatter = getEventsTableColumnDefs()[5].filterValueGetter;
    if (typeof lonDefFormatter === 'function') {
      testValue = lonDefFormatter(formatterParams);
    }
    expect(testValue).toMatch(`${location.longitudeDegrees}`);

    // mb
    const mbDefFormatter = getEventsTableColumnDefs()[8].filterValueGetter;
    if (typeof mbDefFormatter === 'function') {
      testValue = mbDefFormatter(formatterParams);
    }
    expect(testValue).toMatch('5.2');

    // ms
    const msDefFormatter = getEventsTableColumnDefs()[9].valueFormatter;
    if (typeof msDefFormatter === 'function') {
      testValue = msDefFormatter(formatterParams);
    }
    expect(testValue).toMatch('4.9');

    // ml
    const mlDefFormatter = getEventsTableColumnDefs()[10].filterValueGetter;
    if (typeof mlDefFormatter === 'function') {
      testValue = mlDefFormatter(formatterParams);
    }
    expect(testValue).toMatch('5.0');
  });
});
