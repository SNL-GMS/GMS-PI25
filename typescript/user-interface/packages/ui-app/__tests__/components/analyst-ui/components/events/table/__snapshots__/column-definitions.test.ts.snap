// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Table Column Defs has column definitions 1`] = `
[
  {
    "cellRenderer": [Function],
    "comparator": [Function],
    "field": "unsavedChanges",
    "headerClass": "ag-grid__unsaved-changes",
    "headerComponent": {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "type": [Function],
    },
    "headerName": "Unsaved changes",
    "headerTooltip": "Unsaved changes",
    "width": 55,
  },
  {
    "cellRenderer": [Function],
    "field": "conflict",
    "headerName": "Conflict",
    "headerTooltip": "Conflict",
  },
  {
    "field": "time",
    "headerName": "Time",
    "headerTooltip": "Time",
    "sort": "asc",
    "valueFormatter": [Function],
    "valueGetter": [Function],
    "width": 300,
  },
  {
    "field": "timeUncertainty",
    "headerName": "Time Std Deviation (s)",
    "headerTooltip": "Event time standard deviation",
    "valueFormatter": [Function],
    "valueGetter": [Function],
    "width": 150,
  },
  {
    "field": "latitudeDegrees",
    "filterValueGetter": [Function],
    "headerName": "Lat (°)",
    "headerTooltip": "Latitude in degrees",
    "valueFormatter": [Function],
  },
  {
    "field": "longitudeDegrees",
    "filterValueGetter": [Function],
    "headerName": "Lon (°)",
    "headerTooltip": "Longitude in degrees",
    "valueFormatter": [Function],
  },
  {
    "field": "depthKm",
    "headerName": "Depth (km)",
    "headerTooltip": "Depth in kilometers",
    "valueFormatter": [Function],
    "valueGetter": [Function],
    "width": 200,
  },
  {
    "field": "depthUncertainty",
    "headerName": "Depth Std Deviation",
    "headerTooltip": "Depth standard deviation",
    "valueFormatter": [Function],
    "valueGetter": [Function],
    "width": 150,
  },
  {
    "comparator": [Function],
    "field": "magnitudeMb",
    "filterValueGetter": [Function],
    "headerName": "mb",
    "headerTooltip": "Body wave magnitude",
    "valueFormatter": [Function],
    "width": 60,
  },
  {
    "comparator": [Function],
    "field": "magnitudeMs",
    "filterValueGetter": [Function],
    "headerName": "ms",
    "headerTooltip": "Surface wave magnitude",
    "valueFormatter": [Function],
    "width": 60,
  },
  {
    "comparator": [Function],
    "field": "magnitudeMl",
    "filterValueGetter": [Function],
    "headerName": "ml",
    "headerTooltip": "Local magnitude",
    "valueFormatter": [Function],
    "width": 60,
  },
  {
    "field": "numberAssociated",
    "headerName": "# Associated",
    "headerTooltip": "Number of associated signal detections",
    "valueFormatter": [Function],
    "width": 140,
  },
  {
    "field": "numberDefining",
    "headerName": "# Defining",
    "headerTooltip": "Number of time defining signal detections",
    "valueFormatter": [Function],
    "width": 125,
  },
  {
    "field": "observationsStandardDeviation",
    "headerName": "Sdobs",
    "headerTooltip": "Standard deviation of observations",
    "valueFormatter": [Function],
  },
  {
    "field": "coverageSemiMajorAxis",
    "headerName": "Coverage semi-major",
    "headerTooltip": "Coverage semi-major axis in kilometers",
    "valueFormatter": [Function],
    "width": 130,
  },
  {
    "field": "coverageSemiMinorAxis",
    "headerName": "Coverage semi-minor",
    "headerTooltip": "Coverage semi-minor axis in kilometers",
    "valueFormatter": [Function],
    "width": 130,
  },
  {
    "field": "confidenceSemiMajorAxis",
    "headerName": "Confidence semi-major",
    "headerTooltip": "Confidence semi-major axis in kilometers",
    "valueFormatter": [Function],
    "width": 130,
  },
  {
    "field": "confidenceSemiMinorAxis",
    "headerName": "Confidence semi-minor",
    "headerTooltip": "Confidence semi-minor axis in kilometers",
    "valueFormatter": [Function],
    "width": 130,
  },
  {
    "field": "region",
    "headerName": "Region",
    "headerTooltip": "Region",
  },
  {
    "field": "activeAnalysts",
    "headerName": "Active analysts",
    "headerTooltip": "Active analysts",
    "tooltipValueGetter": [Function],
    "valueFormatter": [Function],
    "valueGetter": [Function],
    "width": 200,
  },
  {
    "field": "preferred",
    "headerName": "Preferred in stage",
    "headerTooltip": "Preferred in stage",
    "valueFormatter": [Function],
    "width": 115,
  },
  {
    "cellRendererFramework": [Function],
    "field": "status",
    "headerName": "Workflow status",
    "headerTooltip": "Workflow status",
    "width": 110,
  },
  {
    "field": "rejected",
    "headerName": "Rejected",
    "headerTooltip": "Rejected",
    "valueFormatter": [Function],
    "width": 110,
  },
  {
    "field": "deleted",
    "headerName": "Deleted",
    "headerTooltip": "Deleted",
    "valueFormatter": [Function],
  },
]
`;
